### PR TITLE
have mirror sockets bind to 0.0.0.0 and accept incoming packets only from the browser

### PR DESCRIPTION
**Pull request by trevj**
Tested with uProxy, one peer running Chrome and the other Firefox.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/196)

### CODE REVIEW

**Comment by bemasc** [comment]
Isn't this unsafe, without any source filtering?

**Comment by trevj** [comment]
Just doing one thing at a time.

**Comment by bemasc** [comment]
Introducing a known security-ish issue into a point release, with the intention of fixing it in the following one, makes me very uncomfortable.  I would prefer to do these changes in the other order.

**Comment by trevj** [comment]
This was a lot easier than I expected!

PTAL. Tested with uProxy between Chrome and Firefox and Simple SOCKS on both Chrome and Firefox and manually sending packets to a mirror socket with ncat.

File: src/churn-pipe/churn-pipe.ts [hunk 1]
```diff
@@ -164,13 +164,34 @@ class Pipe {
...
-        this.sendTo_(recvFromInfo.data, remoteEndpoint);
+        // Ignore packets that do not originate from the browser, for a
+        // theoretical security benefit.
+        if (!this.browserEndpoint_) {
+          log.warn('browser endpoint not set, mirror socket for %1 ignoring ' +
+              'incoming packet from %2', remoteEndpoint, {
+                address: recvFromInfo.address,
+                port: recvFromInfo.port
+              });
+        } else if (recvFromInfo.address !== this.browserEndpoint_.address &&
```
Comment (bemasc) [inline src/churn-pipe/churn-pipe.ts]: `||`

**Comment by bemasc** [comment]
:+1:

File: src/churn-pipe/churn-pipe.ts (hunk 1 above)
Comment (trevj) [inline src/churn-pipe/churn-pipe.ts]: Oops! Fixed. Thanks.

**Pull request merged**